test/distributed/optim/test_zero_redundancy_optimizer.py

```diff
@@ -482,6 +450,113 @@ def closure():
             )
             check(optimizer)
 
+    @skip_if_no_gpu
+    def test_state_dict_distributed(self):
```

Comment:
add more checks on the sharded test dict, test reproductibility (run, checkpoint, run A, reload checkpoint, run B, check that run A and run B are the same)